lib/server.js

```diff
@@ -36,22 +40,19 @@ const initServer = function(
     this._cachedVersions.set(appName, versions);
   });
 
-  if (typeof(authPolicy) === 'number') {
-    heartbeatInterval = authPolicy;
-    authPolicy = null;
-  }
-
   this.applications = applications;
   this.heartbeatInterval = heartbeatInterval;
 
-  if (typeof authPolicy === 'function') {
-    this.startSession = authPolicy;
-  } else {
-    if (!authPolicy) authPolicy = new SimpleAuthPolicy();
-    this.startSession = authPolicy.startSession.bind(authPolicy);
+  if (typeof authPolicy === 'object') {
+    authPolicy = authPolicy.authenticate.bind(authPolicy);
```

Comment:
Maybe rename `authPolicy` to `authenticate`? It's usage in code will be clearer in my opinion.